### DISCUSSION TITLE
Get domain of variables after each iteration

### DISCUSSION THREAD

**Discussion by fstrucktru** [google/or-tools#3701] [CP-SAT questions] [answered]
If you run the job shop scheduling example (https://developers.google.com/optimization/scheduling/job_shop) with `solver.paramters.log_search_progress = True`, it will tell you there are 17 variables in the inital model and nine after presolve. For those nine variables, it prints out how many are in which domain: 
```py
#Variables: 9 (#ints:1 in objective)
  - 3 in [0,14]
  - 1 in [2,16]
  - 2 in [3,17]
  - 1 in [4,18]
  - 1 in [5,19]
  - 1 in [9,21]
```

This list is almost what I want to know, but additionally I want to know, which variable it is specifically. Also, I'm not only interested in these values after presolve, but after each solver iteration. What is the best way to do that?

**Comment by lperron** [comment, marked as answer]
there is a parameter to fill in the presolved domains: fill_tightened_domains_in_response along with stop_after_presolve.

There is no programmatic way to get more than this information.

**Reply by fstrucktru** [reply]
Thanks!
